<?xml version="1.0"?>
<launch>
    <arg name="behavior" default="true" description="Whether the behavior control system should be started" />
    <arg name="behavior_dsd_file" default="main.dsd" description="The behavior dsd file that should be used" />
    <arg name="fake_localization" default="true" description="Whether to use the fake localizator"/>
    <arg name="test_series" default="straight_goal" description="Which tests to run"/>
    <arg name="only_script" default="false"/>
    <arg name="no_script" default="false"/>
    <arg name="headless" default="false"/>

    <let name="use_vision" value="false" if="$(var fake_localization)"/>
    <let name="use_vision" value="true" unless="$(var fake_localization)"/>

    <node pkg="bitbots_auto_test" exec="auto_test" output="screen" unless="$(var no_script)" on_exit="shutdown">
        <param name="fake_localization" value="$(var fake_localization)"/>
        <param from="$(find-pkg-share bitbots_auto_test)/config/$(var test_series).yaml"/>
    </node>

    <!-- load the general simulator -->
    <include file="$(find-pkg-share bitbots_webots_sim)/launch/simulation.launch" unless="$(var only_script)">
        <arg name="fake_localization" value="$(var fake_localization)" />
        <arg name="fake_localization_ball" value="$(var fake_localization)" />
        <arg name="camera" value="$(var use_vision)" />
        <arg name="headless" value="$(var headless)" />
    </include>

    <!-- load teamplayer software stack -->
    <include file="$(find-pkg-share bitbots_bringup)/launch/teamplayer.launch" unless="$(var only_script)">
        <arg name="behavior" value="$(var behavior)" />
        <arg name="behavior_dsd_file" value="$(var behavior_dsd_file)" />
        <arg name="game_controller" value="false"/>
        <arg name="ipm" value="$(var use_vision)" />
        <arg name="localization" value="$(var use_vision)" />
        <arg name="motion" value="true" />
        <arg name="sim" value="true" />
        <arg name="path_planning" value="true" />
        <arg name="teamcom" value="false" />
        <arg name="vision" value="$(var use_vision)"/>
        <arg name="world_model" value="$(var use_vision)"/>
    </include>

    <include file="$(find-pkg-share bitbots_auto_test)/launch/rosbag_record_test.launch.py" unless="$(var only_script)">
    </include>

</launch>
